serverless/serverless
# Fix restApiId resolution in post CF deployment phase

## What did you implement:

Fixed resolution of `restApiId` in post CF deployment phase

Closes #6100 

## How did you implement it:

- Custom `restApiId` as set in `serverless.yml` where not taken into account. Ensure it's not the case
- Ensure `restApiId` is resolved as expected even if user has over 500 API's deployed.

## How can we verify it:

1. Create APIGateway manually (e.g. through AWS Console)
2. Create a project that refers to that APIGateway (via `restApiId` and `restApiRootResourceId` settings) and adds some new HTTP endpoints
3. Confirm project deploys without an error and that HTTP endpoint is created

## Todos:


***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO
